File: templates/solver.cpp
#include <iostream>

#include "solver.hpp"

/*
replace the type parameters for AocSolver such that the first parameter
represents the type of the parsed data and the second one the type of
the solutions
*/
using Solver = AocSolver<std::string, int>;

template<>
void Solver::parse_data() {
    // parse this->raw_data into this->parse_data
    return;
}

template<>
int Solver::solve1() {
    return 0;
}

template<>
int Solver::solve2() {
    return 0;
}


int main() {
    Solver solver = Solver();
    solver.read_input("input.txt");
    solver.parse_data();
    
    std::cout << "Part 1: " << solver.solve1() << std::endl;
    std::cout << "Part 1: " << solver.solve2() << std::endl;
    return 0;
}

File: templates/jl/Solver.jl
module Solver
using Test
using AoC
using AoC.Utils


function parse_input(raw_data)
    raw_data
end
export parse_input


function solve1(parsed)
end
export solve1


function solve2(parsed)
end
export solve2


solution = Solution(parse_input, solve1, solve2)

testinput = """
"""
testanswer_1 = nothing
testanswer_2 = nothing
export testinput, testanswer_1, testanswer_2

test() = AoC.test_solution(solution, testinput, testanswer_1, testanswer_2)
export test

main(part=missing) = AoC.main(solution, part)
export main


end # module Solver
